builtin/providers/aws/resource_aws_dc_intra_virtual_interface_confirm.go

```diff
@@ -0,0 +1,216 @@
+		}
+
+	}
+
+	var stateConf *resource.StateChangeConf
+
+	if v, ok := d.GetOk("allow_down_state"); ok && v.(bool) {
+
+		stateConf = &resource.StateChangeConf{
+			Pending:    []string{"pending"},
+			Target:     []string{"available", "down"},
+			Refresh:    DirectConnectIntraVirtualInterfaceConfirmRefreshFunc(conn, d.Get("virtual_interface_id").(string)),
+			Timeout:    10 * time.Minute,
+			Delay:      10 * time.Second,
+			MinTimeout: 10 * time.Second,
+		}
+
+	} else {
+
+		stateConf = &resource.StateChangeConf{
+			Pending:    []string{"pending", "down"},
+			Target:     []string{"available"},
+			Refresh:    DirectConnectIntraVirtualInterfaceConfirmRefreshFunc(conn, d.Get("virtual_interface_id").(string)),
+			Timeout:    30 * time.Minute,
+			Delay:      10 * time.Second,
+			MinTimeout: 10 * time.Second,
+		}
+
+	}
+
+	_, stateErr := stateConf.WaitForState()
+	if stateErr != nil {
+		return fmt.Errorf(
+			"Error waiting for DirectConnect PrivateVirtualInterface (%s) to become ready: %s",
+			d.Get("virtual_interface_id").(string), err)
+	}
+
+	d.SetId(d.Get("virtual_interface_id").(string))
+
+	// Read off the API to populate our RO fields.
+	return resourceAwsDirectConnectIntraVirtualInterfaceConfirmRead(d, meta)
+}
+
+func DirectConnectIntraVirtualInterfaceConfirmRefreshFunc(conn *directconnect.DirectConnect, virtualinterfaceId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
+			VirtualInterfaceId: aws.String(virtualinterfaceId),
+		})
+
+		if err != nil {
+
+			log.Printf("Error on DirectConnectPrivateVirtualInterfaceRefresh: %s", err)
+			return nil, "", err
+
+		}
+
+		if resp == nil || len(resp.VirtualInterfaces) == 0 {
+			return nil, "", nil
+		}
+
+		virtualInterface := resp.VirtualInterfaces[0]
+		return virtualInterface, *virtualInterface.VirtualInterfaceState, nil
+	}
+}
+
+func resourceAwsDirectConnectIntraVirtualInterfaceConfirmRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dcconn
+
+	resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
+		VirtualInterfaceId: aws.String(d.Id()),
+	})
+
+	if err != nil {
+
+		log.Printf("[ERROR] Error finding DirectConnect PrivateVirtualInterface: %s", err)
+		return err
+
+	}
+
+	if len(resp.VirtualInterfaces) != 1 {
+		return fmt.Errorf("[ERROR] Error finding DirectConnect PrivateVirtualInterface: %s", d.Id())
+	}
+
+	virtualInterface := resp.VirtualInterfaces[0]
+
+	// Set attributes under the user's control.
+	d.Set("connection_id", *virtualInterface.ConnectionId)
+	d.Set("asn", *virtualInterface.Asn)
+	d.Set("virtual_interface_name", *virtualInterface.VirtualInterfaceName)
+	d.Set("vlan", *virtualInterface.Vlan)
+	d.Set("amazon_address", *virtualInterface.AmazonAddress)
+	d.Set("customer_address", *virtualInterface.CustomerAddress)
+	// d.Set("auth_key", *virtualInterface.AuthKey)
```

Comment:
I don't remember why I made that choice back then... it has been a while, but I suspect it was because I was trying to avoid printing the AuthKey. Since any change to a VIF forces a recreate as there is no update method it should not matter, unless I am mistaken and causes an issue with detecting a change in the AuthKey.